WireUI/Sources/WireReusableUIComponents/AccountImageView/AccountImageView.swift

```diff
@@ -0,0 +1,330 @@
+    }
+
+    private func updateAvailabilityIndicator() {
+
+        if availabilityIndicatorView.availability != availability {
+            availabilityIndicatorView.availability = availability
+        }
+
+        // for dark mode
+        availabilityIndicatorBackgroundView.isHidden = availability == .none || traitCollection.userInterfaceStyle != .dark
+
+        if availability == .none || traitCollection.userInterfaceStyle == .dark {
+            // remove clipping
+            accountImageView.superview?.layer.mask = .none
+            return
+        }
+
+        // draw a rect over the total bounds (for inverting the arc)
+        let imageWrapperViewHeight = accountImageBorderWidth * 2 + accountImageHeight
+        let maskPath = UIBezierPath(
+            rect: .init(origin: .zero, size: .init(width: imageWrapperViewHeight, height: imageWrapperViewHeight))
+        )
+        // crop a circle shape from the image
+        let center = CGPoint(x: availabilityIndicatorCenterOffset, y: availabilityIndicatorCenterOffset)
+        let radius = availabilityIndicatorRadius + availabilityIndicatorBorderWidth
+        maskPath.addArc(withCenter: center, radius: radius, startAngle: 0, endAngle: 2 * .pi, clockwise: true)
+
+        // this clips a circle from the view, which gives the
+        // availability indicator view a transparent border
+        let maskLayer = CAShapeLayer()
+        maskLayer.path = maskPath.cgPath
+        maskLayer.fillRule = .evenOdd
+        accountImageView.superview?.layer.mask = maskLayer
+    }
+
+    // MARK: - Nested Types
+
+    public enum AccountType: CaseIterable {
+        /// The account image will be clipped using a circle shape.
+        case user
+        /// The account image will be clipped using a round rectangle shape.
+        case team
+    }
+}
+
+// MARK: - Convenience Init
+
+extension AccountImageView {
+
+    public convenience init(
+        accountImage: UIImage,
+        accountType: AccountType,
+        availability: Availability?
+    ) {
+        self.init()
+
+        self.accountImage = accountImage
+        self.accountType = accountType
+        self.availability = availability
+
+        updateAccountImage()
+        updateShape()
+        updateAvailabilityIndicator()
+    }
+}
+
+// MARK: - Previews
+
+@available(iOS 16.0, *)
+struct AccountImageView_Previews: PreviewProvider {
+
+    static var previews: some View {
+        Group {
+            ForEach(AccountImageView.AccountType.allCases, id: \.self) { accountType in
+
+                previewWithNavigationBar(accountType, .none)
+                    .previewDisplayName("\(accountType)")
+
+                ForEach(Availability.allCases, id: \.self) { availability in
+                    previewWithNavigationBar(accountType, availability)
+                        .previewDisplayName("\(accountType) - \(availability)")
+                }
+            }
+        }
+        .background(Color(UIColor.systemGray2))
+    }
```

Comment:
Like we discussed we can simplify Previews by using the #Preview macro. Which is available from iOS 17 and above.

Reply:
I changed the code a bit to use the `#Preview` macro, but I wasn't able to make it more compact or get rid of the `UIViewRepresentable`.

Reply:
Looking good!